File: cognitodemo/confirm.js
const Cognito = require("@aws-sdk/client-cognito-identity-provider");
const crypto = require("crypto");

const clientId = "";  // Obtain from the AWS console
const clientSecret = "";  // Obtain from the AWS console
const username = "theuser";
const confirmationCode = ""; // obtain from your email

function secretHash(clientId, clientSecret, username) {
  const hasher = crypto.createHmac('sha256', clientSecret);
  hasher.update(`${username}${clientId}`);
  return hasher.digest('base64');
}


async function main() {
    const client = new Cognito.CognitoIdentityProviderClient({ region: 'ap-southeast-2' });
  const command2 = new Cognito.ConfirmSignUpCommand({
    ClientId: clientId,
    SecretHash: secretHash(clientId, clientSecret, username),
    Username: username,
    ConfirmationCode: confirmationCode,
  });

  res2 = await client.send(command2);
  console.log(res2);

}

main();

File: cognito_python/confirm.py
import boto3

import hmac, hashlib, base64 

username= "testuser"
password= "MySuperSecret99!"
email= "Your email goes here"
client_id = ""  # Obtain from the AWS console
client_secret = ""  # Obtain from the AWS console
confirmation_code = "" #Obtain from the verification email

def secretHash(clientId, clientSecret, username):
    message = bytes(username + clientId,'utf-8') 
    key = bytes(clientSecret,'utf-8') 
    return base64.b64encode(hmac.new(key, message, digestmod=hashlib.sha256).digest()).decode() 

def confirm():
    client = boto3.client("cognito-idp", region_name="ap-southeast-2")
    try:
        response = client.confirm_sign_up(
            ClientId=client_id,
            Username=username,
            ConfirmationCode=confirmation_code,
            SecretHash=secretHash(client_id, client_secret, username)
        )
        return response
    except Exception as e:
        print(f"Error during confirmation: {e}")
        return None

confirm_response = confirm()
print("Confirmation successful:", confirm_response)